cmd/juju/cloud/updatecredential.go

```diff
@@ -17,31 +17,31 @@ import (
 )
 
 var usageUpdateCredentialSummary = `
-Updates a controller credential for a cloud.`[1:]
+Updates a remote credential.`[1:]
 
 var usageUpdateCredentialDetails = `
-Cloud credentials for controller are used for model operations and manipulations.
-Since it is common to have long-running models, it is also common to 
-have these cloud credentials become invalid during models' lifetime.
-When this happens, a user must update the cloud credential that 
-a model was created with to the new and valid details on controller.
-
-This command allows to update an existing, already-stored, named,
-cloud-specific controller credential.
-
-NOTE: 
-This is the only command that will allow you to manipulate cloud
-credential for a controller. 
-All other credential related commands, such as 
-` + "`add-credential`" + `, ` + "`remove-credential`" + ` and  ` + "`credentials`" + ` 
-deal with credentials stored locally on the client not on the controller.
+This command updates a credential cached on a controller. It does this by
+uploading the identically named local credential and validating it against the
```

Comment:
What? No it does not and even it did, it makes no sense to say so. How is it relevant to the usage?

Reply:
It *looks* like there is validation that the new credential is the same as the credential stored in its models. So, if this is the case, my bad, not cloud validation, but model validation. I said cloud because the error text is unclear:

```
Credential invalid for:
  pete-model:
    
The provided credentials could not be validated and 
may not be authorized to carry out the request.
Ensure that your account is authorized to use the Amazon EC2 service and 
that you are using the correct access keys. 
These keys are obtained via the "Security Credentials"
page in the AWS console.
: AWS was not able to validate the provided access credentials (AuthFailure)
Controller credential "pete" for user "admin" on cloud "aws" not updated: some models are no longer visible.
```

In passing, what is one supposed to do when the command errors out like this, which would be almost all the time?

Reply:
Before we update the cloud credential content, we validate that it works for all the models that refer to it. If the credential succeeds/is valid for all models that use it, then its contents are updated on the controller. The message will list what models it was validated against.

If the model validation fails for even one of the models that use this credential, the update of the credential content will not happen. We will list the models that failed and the error, they would have failed with. In you r case, the credential content was still invalid and was not passing basic auth. There could be scenarios where the model references some machine instances but the new credential does not see them or vice versa - the instances are visible but Juju is not aware of them.

When  model validation for a new credential fails, depending on what error the model has, the user can consider:
(i) choosing different credential content and using update-credential again; 
(ii) choosing different credential for the failing model and using set-crdential;
(iii) if the issue is with machines/instances, addressing that, i.e. removing machine, etc (note that force remove machine is yet to come)...